useful-scripts/importer-tests/index.ts

```diff
@@ -0,0 +1,203 @@
+      },
+    });
+    console.timeEnd('Create-release-time');
+    const releaseId = res.data.id;
+    console.log(
+      `Release URL: ${API_URL}/publication/${publicationId}/release/${releaseId}/data`,
+    );
+    await FileOps(releaseId);
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+const FileOps = async (releaseId: string) => {
+  const zippedFile = await globby('*.zip');
+  const cleanZipFile = zippedFile[0];
+  const archive = fs.existsSync(zippedFile[0]);
+  console.log('archive', archive);
+  if (archive) {
+    const zip = new StreamZip.async({ file: cleanZipFile });
+    const count = await zip.extract(null, './test-files');
+    console.log(`Extracted ${count} entries to test-files`);
+    await zip.close();
+  }
+
+  // rename csv file..
+  const csvGlob1 = await globby('./test-files/*.csv');
+  const csvGlob2 = csvGlob1[0];
+  const randomId = Math.floor(Math.random() * 1000000);
+
+  fs.rename(`${csvGlob2}`, `./test-files/testfile-${randomId}.csv`, e =>
+    e ? console.error(e) : '',
+  );
+
+  // rename the meta file...
+  const metaGlob = await globby('./test-files/*.meta.csv');
+  const metaGlob2 = metaGlob[0];
+
+  fs.rename(`${metaGlob2}`, `./test-files/testfile.meta.${randomId}.csv`, e =>
+    e ? console.error(e) : '',
+  );
+  await ZipDirectory('./test-files', `./zip-files/clean-test-zip-${v4()}.zip`);
+  await addSubject(releaseId);
+};
+
+const addSubject = async (releaseId: string) => {
+  const glob = await globby('./zip-files/*.zip');
+  if (glob !== undefined) {
+    const oldPath =
+      __dirname + '\\zip-files\\' + fs.readdirSync(__dirname + '/zip-files')[0];
+    const newPath =
+      __dirname + '\\' + fs.readdirSync(__dirname + '/zip-files')[0];
+    fs.renameSync(oldPath, newPath);
+  }
+  const rootGlob = await globby('clean-test-zip-*.zip');
+  const form = new FormData();
+  form.append('zipFile', fs.createReadStream(rootGlob[0]));
+
+  try {
+    const res = await axios({
+      maxContentLength: Infinity,
+      maxBodyLength: Infinity,
+      method: 'POST',
+      url: `${API_URL}/api/release/${releaseId}/zip-data?name=importer-subject=${v4()}`,
+      data: form,
+      headers: {
+        ...form.getHeaders(),
+        Authorization: `Bearer ${JWT_TOKEN}`,
+      },
+    });
+    const subjectId = res.data.id;
+    await getSubjectProgress(releaseId, subjectId);
+    console.log(res.data);
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+const getSubjectProgress = async (releaseId: string, subjectId: string) => {
+  console.time('import subject upload time');
+  try {
+    const res = await axios({
+      method: 'GET',
+      url: `${API_URL}/api/release/${releaseId}/data/${subjectId}/import/status`,
+      headers: {
+        Authorization: `Bearer ${JWT_TOKEN}`,
+        'Content-Type': 'application/json',
+      },
+    });
+    if (res.data.status !== 'COMPLETE') {
+      console.log(res.data);
+      await Sleep(300);
+      await getSubjectProgress(releaseId, subjectId);
```

Comment:
Rather than making getSubjectProgress recursive, I'd create a loop at the bottom of the file so that it looks something like this

```
var pubId = createPub();
var releaseId = createRelease();
FileOps(releaseId);
ZipDirectory(blah);
var subId = addSubject(releaseId);
var subComplete = false;
while(!subComplete) { subComplete = getSubjectProgress(releaseId, subId); }
```

So getSubjectProgress can just return the progress itself.